### PR TITLE
Issue 1218/contact card

### LINKED ISSUE #1218: 🤸‍♀️ Event contact assignment (card and from list)
## Description
A user should be able to assign a "contact person" for an event, that is the person in charge of staying in touch with all participants, and making the necessary preparations for the event to take place.

Also see epic #1214 for context.

## Relevant Job Stories
* When I’m using events to organize basic activism like handing out flyers or putting up posters, I want to make sure that at least one of the people signed up for the event has done the work before and assign them as the contact for the rest of the activists, so that I can feel comfortable that they will be able to run the event
* When I’m using events to organize basic activism and one of the actvists who signed up would be suitable for running the event, I want to reach out to them, so that I can make sure they are able to do so
* When I’m using events to organize basic activism and I have done all of the preparations (assigning a contact etc), I want to be able to send out a reminder e-mail, so that all of the attendees get the information they need

## Prerequisites
Don't start working on this until #1215 is merged.

## Requirements
- [x] There should be a card at the top right of the Participants page
- [x] When there is no contact person assigned:
  - [x] The card should have the title "No contact assigned"
  - [x] There should be a `PersonSelectWidget` to pick a contact person
- [x] When there is a contact person assigned:
  - [x] The card should have the title "Contact"
  - [x] The avatar and name of the contact should be displayed
  - [x] There should be a "Remove" button, that when clicked should open a confirmation modal informing that the contact will be remove but still a participant
- [x] In the list of participants, the currently assigned contact should have a small icon and the text "Contact person"
- [x] In the list of participants, hovering any person except the current contact person should show a faded icon with the text "Make contact person"

## Possible implementations
Use the `showConfirmDialog` from the `ZUIConfirmDialogContext`.

### In situ on page
image

The card in question is the one marked by a purple border below.
image

### No contact assigned
image

### Contact assigned
image

### Assigning contact person in list
The screenshot below is missing the text, that should be displayed next to the icon. The second person in the mockup is the current contact person. The third person in the list is hovered.

image

## Open questions
None

### PR DESCRIPTION
**Pull request by rebecarubio**
## Description
This PR adds the Event Contact Card in the participant page to select an event contact.

## Screenshots
https://user-images.githubusercontent.com/36491300/235176935-8f2c5532-2b63-4744-8686-fd843d089fbd.mp4

## Changes
* Adds Event contact card
* Adds icon in participants list to show which participant is the assigned contact
* Adds on hover a participant to set it as a contact

## Notes to reviewer
None

## Related issues
Resolves #1218

### CODE REVIEW

File: src/pages/organize/[orgId]/projects/[campId]/events/[eventId]/participants.tsx [hunk 1]
```diff
@@ -53,6 +54,13 @@ const ParticipantsPage: PageWithLayout<ParticipantsProps> = ({
               <Grid item md={8} xs={12}>
                 <ParticipantSummaryCard model={dataModel} />
               </Grid>
+              <Grid item md={4} xs={8}>
```
Comment (richardolsson) [inline src/pages/organize/[orgId]/projects/[campId]/events/[eventId]/participants.tsx:57]: I think `xs` should be `12`, so that it occupies the full width on small screens.

File: src/features/events/components/EventContactCard.tsx [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+                onSubmit: () => {
+                  model.removeContact();
+                },
+                warningText: messages.eventContactCard.warningText(),
+              });
+            }}
+            size="small"
+            variant="text"
+          >
+            <CloseIcon />
```
Comment (richardolsson) [inline src/features/events/components/EventContactCard.tsx:120]: When you want an icon in a button, you can (and should) use the `startIcon` prop, like so:

```tsx
<Button startIcon={<Close/>}>Label here</Button>
```

☝️  I'm writing this from memory so I may have made a mistake somewhere, but I think the idea comes across.

File: src/features/events/components/EventContactCard.tsx [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+                <Box
+                  m={1}
+                  mb={3}
+                  mt={3}
+                  sx={{
+                    '& input': {
+                      alignSelf: 'self-start',
+                      minWidth: '100px !important',
+                      width: '200px !important',
+                    },
```
Comment (richardolsson) [inline src/features/events/components/EventContactCard.tsx:90]: Why is this necessary? I'm guessing that this is what's causing the person select widget to be too wide on narrow screens (see screenshot). This seems like a hack to me, but I understand that there was probably some issue without it that you wanted to solve. I'm curious what that was, maybe we can find a simpler solution?

image

File: src/features/events/components/EventContactCard.tsx [hunk 4]
```diff
@@ -0,0 +1,129 @@
+import CloseIcon from '@mui/icons-material/Close';
+import FaceOutlinedIcon from '@mui/icons-material/FaceOutlined';
+import FaceRetouchingOffOutlinedIcon from '@mui/icons-material/FaceRetouchingOffOutlined';
```
Comment (richardolsson) [inline src/features/events/components/EventContactCard.tsx:3]: Instead of importing these from their respective single-icon modules, import them all from `@mui/icons-material`, like so:

```tsx
import { Close, FaceOutlined, FaceRetouchingOff } from '@mui/icons-material`;
```

File: src/features/events/components/EventContactCard.tsx [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+                >
+                  <ZUIAvatar orgId={orgId} personId={data.contact.id} />
+                </Box>
+                <Typography
+                  sx={{ display: 'inline-block', verticalAlign: 'middle' }}
+                >
+                  {data.contact.name}
+                </Typography>
+              </>
+            ) : (
```
Comment (richardolsson) [inline src/features/events/components/EventContactCard.tsx:79]: I'm not a fan of these long ternary expressions because I find it difficult to see where the expressions start and end. Can it be refactored, e.g. as two components (defined in the same module), something like this:

```tsx
{!!data.contact ? (
  <ContactDetails contact={data.contact} />
) : (
  <ContactSelect onSelect={/*...*/} />
)}
```

Or using functions (that can be defined higher up in the component function):

```tsx
{!!data.contact ? (
  renderContactDetails(data.contact)
) : (
  renderContactSelect()
)}
```

**Review by richardolsson — CHANGES_REQUESTED** [review]
Awesome work on this. So fast! 🚀 

I've found a couple of minor things that I would like to change, mostly code related but also some related to functionality/usability. Have a look when you get a chance! 👇

File: src/features/events/components/EventContactCard.tsx (hunk 3 above)
Comment (rebecarubio) [reply src/features/events/components/EventContactCard.tsx:90]: Yes! The input from the `ZUIPersonSelect `has by default a _min-width: 30px_ so this was the only way I found to make the placeholder display the content and override the default min-width, otherwise:

image

But to avoid this behavior I can have a min-width of 100px for example to don't let the input be cutted?

Comment (richardolsson) [reply src/features/events/components/EventContactCard.tsx:92]: Lets find some time tomorrow to look at this together. It should not be possible to employ hacks like this. Instead, lets find a nice way to control the min width from outside the component (if at all needed).

File: src/features/events/components/EventContactCard.tsx [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+      </Typography>
+      <Box m={1}>
+        <Button
+          onClick={() => {
+            showConfirmDialog({
+              onSubmit: () => {
+                model.removeContact();
+              },
+              warningText: messages.eventContactCard.warningText({
+                name: contact!.name,
```
Comment (richardolsson) [inline src/features/events/components/EventContactCard.tsx:54]: The "non-null assertion" operator (`!`) should be avoided, and I think it's not even needed in this case (because the type of `contact` does not allow `null`).

File: src/features/events/components/EventContactCard.tsx [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+};
+
+const ContactSelect: FC<ContactSelectProps> = ({ model }) => {
+  const messages = useMessages(messageIds);
+  const handleSelectedPerson = (personId: number) => {
+    model.setContact(personId);
+  };
+
+  return (
+    <Box sx={{ display: 'grid' }}>
```
Comment (richardolsson) [inline src/features/events/components/EventContactCard.tsx:76]: The use of `display: 'grid'` surprises me. I see no reason for using a CSS grid here, and the lack of any other grid-related properties makes me think it probably doesn't do much difference either?

File: src/features/events/components/EventContactCard.tsx [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+          <Box>
+            {data.contact ? (
+              <ContactDetails
+                contact={data.contact}
+                model={model}
+                orgId={orgId}
+              />
+            ) : (
+              <ContactSelect model={model} />
+            )}
```
Comment (richardolsson) [inline src/features/events/components/EventContactCard.tsx:147]: Much better! Easier to read. 💯

**Review by richardolsson — CHANGES_REQUESTED** [review]
Nice work on refactoring! I found some more minor things. Lets talk about the styling stuff this afternoon?

File: src/features/events/components/EventContactCard.tsx (hunk 7 above)
Comment (rebecarubio) [reply src/features/events/components/EventContactCard.tsx:76]: You're right, I 'll remove it
